kernel/gic/src/gic/dist_interface.rs

```diff
@@ -208,10 +221,10 @@ impl super::ArmGic {
             };
 
             v3.dist_extended.write_volatile_64(offset::P6IROUTER, value);
+        } else {
+            // If we're on gicv2 then affinity routing is off
+            // so we landed in the first block
+            unreachable!()
```

Comment:
in general, `unreachable` code found in if-else statements is a code smell indicating a design flaw. Is there a better way to represent this to avoid the if-else issue, e.g., a match statement or something?

Reply:
Yep, sure:

```rust
    /// Sets the destination of an SPI.
    pub fn set_spi_target(&mut self, int: InterruptNumber, target: SpiDestination) {
        assert!(int >= 32, "interrupts number below 32 (SGIs & PPIs) don't have a target CPU");
        match self {
            Self::V2(_) | Self::V3(ArmGicV3 { affinity_routing: false, .. }) => {
                if let SpiDestination::Specific(cpu) = &target {
                    assert!(cpu.value() < 8, "affinity routing is disabled; cannot target a CPU with id >= 8");
                }

                let value = match target {
                    SpiDestination::Specific(cpu) => 1 << cpu.value(),
                    SpiDestination::AnyCpuAvailable => {
                        let list = TargetList::new_all_cpus()
                            .expect("This is invalid: CpuId > 8 AND GICv2 interrupt controller");
                        list.0 as u32
                    },
                    SpiDestination::GICv2TargetList(list) => list.0 as u32,
                };

                self.distributor_mut().write_array_volatile::<4>(offset::ITARGETSR, int, value);
            },
            Self::V3(ArmGicV3 { affinity_routing: true, dist_extended, .. }) => {
                let value = match target {
                    SpiDestination::Specific(cpu) => MpidrValue::from(cpu).value(),
                    // bit 31: Interrupt Routing Mode
                    // value of 1 to target any available cpu
                    SpiDestination::AnyCpuAvailable => P6IROUTER_ANY_AVAILABLE_PE,
                    SpiDestination::GICv2TargetList(_) => {
                        panic!("Cannot use SpiDestination::GICv2TargetList with GICv3!");
                    },
                };

                dist_extended.write_volatile_64(offset::P6IROUTER, value);
            }
        }
    }
```

pushing this

Reply:
perfect, well done!